api/user.rs

```diff
@@ -1,23 +1,47 @@
-use id::{wrap_error, oauth_user, db};
-use vercel_runtime::{run, Body, Error, Request, Response};
-use entity::{
-    passport,
-    prelude::*,
-};
+use entity::{passport, prelude::*, sea_orm_active_enums::RoleEnum, user};
+use id::{db, oauth_user, wrap_error};
 use sea_orm::prelude::*;
+use serde::{Deserialize, Serialize};
+use vercel_runtime::{run, Body, Error, Request, Response};
+
+#[derive(Serialize, Deserialize)]
+struct UserWithPassport {
+    id: i32,
+    discord_id: i64,
+    role: RoleEnum,
+    totp: Option<String>,
+    latest_passport: Option<passport::Model>,
+}
 
 #[tokio::main]
 async fn main() -> Result<(), Error> {
     run(wrap_error!(handler)).await
 }
 
 pub async fn handler(req: Request) -> Result<Response<Body>, Error> {
-    let user = oauth_user(req, vec!["user:read".parse().expect("valid scope")]).await?;
+    let user_id = oauth_user(req, vec!["user:read".parse().expect("valid scope")]).await?;
 
     let db = db().await?;
 
-    let passport = Passport::find().filter(passport::Column::OwnerId.eq(user)).one(&db).await?;
+    let user = User::find()
+        .filter(user::Column::Id.eq(user_id))
+        .one(&db)
+        .await?
+        .ok_or_else(|| Error::from("User not found"))?;
+    let latest_passport = Passport::find()
+        .filter(passport::Column::OwnerId.eq(user_id))
+        .one(&db)
+        .await?;
```

Comment:
Do we know that this will always return the latest passport?

Reply:
Idk but this part is unchanged from what it used to return, which was only the latest passport. Maybe it's worth adding an explicit sort for latest?

Reply:
I agree, go ahead and add that. I wrote this in 5 minutes the first time around lol